File: Packages/RagnarokRenderers/Sources/RagnarokShaders/ShadersLibrary.m
//
//  ShadersLibrary.m
//  RagnarokShaders
//
//  Created by Leon Li on 2024/4/10.
//

#import "ShadersLibrary.h"

id<MTLLibrary> RagnarokCreateShadersLibrary(id<MTLDevice> device) {
    NSBundle *bundle = SWIFTPM_MODULE_BUNDLE;
    return [device newDefaultLibraryWithBundle:bundle error:nil];
}

File: Packages/RagnarokRenderers/Sources/RagnarokRenderers/Core/ShadersLibrary.swift
//
//  ShadersLibrary.swift
//  RagnarokRenderers
//
//  Created by Leon Li on 2026/3/23.
//

import Metal
import RagnarokShaders

public func ragnarokShadersLibrary(device: any MTLDevice) -> (any MTLLibrary)? {
    RagnarokCreateShadersLibrary(device)
}
